src/renderers/dom/shared/ReactEventListener.js

```diff
@@ -16,33 +16,38 @@ var ExecutionEnvironment = require('ExecutionEnvironment');
 var PooledClass = require('PooledClass');
 var ReactDOMComponentTree = require('ReactDOMComponentTree');
 var ReactGenericBatching = require('ReactGenericBatching');
+var ReactTypeOfWork = require('ReactTypeOfWork');
 
 var getEventTarget = require('getEventTarget');
 var getUnboundedScrollPosition = require('getUnboundedScrollPosition');
 
+var { HostContainer } = ReactTypeOfWork;
+
 /**
  * Find the deepest React component completely containing the root of the
  * passed-in instance (for use when entire React trees are nested within each
  * other). If React trees are not nested, returns null.
  */
-function findParent(inst) {
+function findRootContainerNode(inst) {
   // TODO: It may be a good idea to cache this to prevent unnecessary DOM
   // traversal, but caching is difficult to do correctly without using a
   // mutation observer to listen for all DOM changes.
-  var container;
   if (typeof inst.tag === 'number') {
     while (inst.return) {
       inst = inst.return;
     }
-    container = inst.stateNode.containerInfo;
+    if (inst.tag !== HostContainer) {
+      // This can happen if we're in a detached tree.
```

Comment:
If this happens, it means that we should throw away the event, right? I don't see that in the code.